aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -114,6 +115,33 @@
    */
   public static final String S3FILEIO_ACL = "s3fileio.acl";
 
+  /**
+   * If {@link org.apache.iceberg.aws.glue.GlueCatalog} should use external lock or not, default to false.
+   * If set to true, it will use DynamoDB to enforce locking during commits.
+   */
+  public static final String GLUE_CATALOG_LOCK_ENABLED = "gluecatalog.lock.enabled";
+  public static final boolean GLUE_CATALOG_LOCK_ENABLED_DEFAULT = true;
```

Comment:
Can the lock have a configurable type? eg DYNAMODB by default? We were hoping to leverage the work here to make nessie an option for the 'locking' mechanism here and bring nessie support to Glue. I hope to have a patch up soon to follow on from this.

Reply:
I'm curious what this would look like.

Reply:
more info in my response to Jack.